documentation/VSTestIntegration.md

```diff
@@ -0,0 +1,92 @@
+|Exclude         | Exclude from code coverage analysing using filter expressions.                           | 
+|ExcludeByFile   | Ignore specific source files from code coverage.                                         | 
+|Include         | Explicitly set what to include in code coverage analysis using filter expressions.       | 
+|IncludeDirectory| Explicitly set which directories to include in code coverage analysis.                   |
+|SingleHit       | Specifies whether to limit code coverage hit reporting to a single hit for each location.| 
+|UseSourceLink   | Specifies whether to use SourceLink URIs in place of file system paths.                  |
+
+How to specify these options via runsettings?
+```
+<?xml version="1.0" encoding="utf-8" ?>
+<RunSettings>
+  <DataCollectionRunSettings>
+    <DataCollectors>
+      <DataCollector friendlyName="XPlat code coverage">
+        <Configuration>
+          <Format>json</Format>
+          <MergeWith>/custom/path/result.json</MergeWith>
+          <Exclude>[coverlet.*.tests?]*,[*]Coverlet.Core*</Exclude> <!-- [Assembly-Filter]Type-Filter -->
+          <Include>[coverlet.*]*,[*]Coverlet.Core*</Include> <!-- [Assembly-Filter]Type-Filter -->
+          <ExcludeByAttribute>Obsolete,GeneratedCodeAttribute,CompilerGeneratedAttribute</ExcludeByAttribute>
+          <ExcludeByFile>../dir1/class1.cs,../dir2/*.cs,../dir3/**/*.cs,</ExcludeByFile> <!-- Absolute or relative file paths -->
+          <IncludeDirectory>../dir1/*,../dir2/,</IncludeDirectory>
+          <SingleHit>false</SingleHit>
+          <UseSourceLink>true</UseSourceLink>
+        </Configuration>
+      </DataCollector>
+    </DataCollectors>
+  </DataCollectionRunSettings>
+</RunSettings>
+```
+This runsettings file can easily be provided using command line option as given :
+
+1. `dotnet test --settings coverletArgs.runsettings`
+
+2. `dotnet vstest --settings coverletArgs.runsettings`
+
+
+#### Scope of Enhancement 
+Currently, advanced options are supported via runsettings. Providing support through additional command line arguments in vstest can be taken up separately.
+
+## Implementation Details
+The proposed solution is implemented with the help of [datacollectors](https://github.com/Microsoft/vstest-docs/blob/master/docs/extensions/datacollector.md). 
+1. Outproc Datacollector : The outproc collector would always run in a separate process(datacollector.exe/datacollector.dll) than the process in which tests are being executed(testhost*.exe/testhost.dll). This datacollector would be responsible for calling into coverlet APIs for instrumenting dlls, collecting coverage results and sending the coverage output file back to test platform.
+2. Inproc Datacollector : The inproc collector in the testhost process executing the tests. This collector will be needed to remove the dependency on the exit handler to flush the hit files.
```

Comment:
I don't see config to plug inproc datacollector...am I missing something?

Reply:
The inproc datacollector will be automatically plugged in with `--collect:"XPlat Code Coverage" `. We have test platform changes for this. The inproc datacollector will lie in the same nuget and get picked from there.

Reply:
ah ok thank's now it's more clear in/out behaviours of my past tests.